linker/Linker/Driver.cs

```diff
@@ -72,7 +74,71 @@ public static int Execute (string[] args, ILogger customLogger = null)
 
 		public Driver (string [] args)
 		{
-			_queue = new Queue<string> (args);
+			_queue = ProcessResponseFile (args);
+		}
+
+		Queue<String> ProcessResponseFile (string [] args)
+		{
+			var result = new Queue<string> ();
+			foreach (string arg in args) {
+				if (arg.StartsWith ("@")) {
+					try
+					{
+						string responseFileName = arg.Substring(1);
+						IEnumerable<string> responseFileLines = File.ReadAllLines(responseFileName).Select(line => line.Trim());
+						ParseResponseFileLines(responseFileLines, result);
+					} catch (Exception) {
+						Usage ("Cannot read response file");
+					}
+				} else {
+					result.Enqueue (arg);
+				}
+			}
+			return result;
+		}
+
+		public static void ParseResponseFileLines (IEnumerable<string> responseFileLines, Queue<string> result)
+		{
+			string responseFileText = string.Join (" ", responseFileLines);
+			int idx = 0;
+			while  (idx < responseFileText.Length) {
+				while  (idx < responseFileText.Length && char.IsWhiteSpace (responseFileText [idx])) {
+					idx++;
+				}
+				if  (idx == responseFileText.Length) {
+					break;
+				}
+				StringBuilder argBuilder = new StringBuilder ();
+				bool inquote = false;
+				while  (true) {
+					bool copyChar = true;
+					int numBackslash = 0;
+					while  (idx < responseFileText.Length && responseFileText [idx] == '\\') {
+						numBackslash++;
+						idx++;
+					}
+					if  (idx < responseFileText.Length && responseFileText [idx] == '"') {
```

Comment:
I don't follow the logic here, are you trying to unquote `"`? Why not do simple one char look ahead?

Reply:
I am trying to be consistent with existing Microsoft tools that take response file as input. The implementation I have is based on some existing implementation, and here is I believe how the rules are derived:

Windows path use the backslash `\` a lot, it would be silly if we have to escape every backslash, so we don't. By default, `\` simply means the literal backslash.

We wanted to have space in file name as well. Conventionally, we use the double quote `"`, so we quote the whole path, all is good.

Then the question arise - how do we process a literal `"` inside an argument? Easy - just escape it with the backslash as we always do like `\"`.

Now we need to fix the problem of producing a literal `\` right before the ending double quote, which is the common case for argument that represents a directory. A plausible solution would be `\\"`. Conceptually, the first backslash escaped the second, so the remaining is just the delimiting double quote, not a literal double quote.

Inductively, we have the problem of producing a literal `\"`, so we choose `\\\"`. Again, conceptually, the first backslash escaped the second, and the third escaped the last double quote, therefore the last double quote is literal, not delimiting.

Noticing the odd even pattern, we come up with these two simple rules:

* If the sequence of backslash does not end with a double quote, treat them as literal backslash.
* Otherwise, we count the number of backslash, generate half (floored if odd) number of backslash, if it is odd, then generate a literal double quote, otherwise generate a delimiting double quote.

In order to implement the rules above, we need to look ahead until we know what follows the backslash sequence, and therefore cannot be a simple one character look ahead.

Granted, the problem of generating these could be much simplified (for example, simply use the line ending as the argument delimiter or use the Roslyn way, using a pair of double quote to mean a literal double quote. Either way the parser is much simplified. I chose this approach mostly to make sure it is consistent with other Microsoft tools.

Reply:
Shall we resolve this?